conformance/utils/kubernetes/helpers.go

```diff
@@ -239,6 +239,24 @@ func NamespacesMustBeReady(t *testing.T, c client.Client, timeoutConfig config.T
 	require.NoErrorf(t, waitErr, "error waiting for %s namespaces to be ready", strings.Join(namespaces, ", "))
 }
 
+// GatewayAcceptedConditionMustEqual requires that a provided Gateway resource includes the Accepted
+// condition present that matches the provided status and reason.
+func GatewayAcceptedConditionMustEqual(t *testing.T, gw *v1beta1.Gateway, status metav1.ConditionStatus, reason v1beta1.GatewayConditionReason) {
+	require.True(t,
+		findConditionInList(t, gw.Status.Conditions, string(v1beta1.GatewayConditionAccepted), string(status), string(reason)),
+		"Accepted status of %s with reason %s was NOT present on Gateway %s/%s as required", status, reason, gw.Namespace, gw.Name,
+	)
+}
+
+// GatewayProgrammedConditionMustEqual requires that a provided Gateway resource includes the Programmed
+// condition present that matches the provided status and reason.
+func GatewayProgrammedConditionMustEqual(t *testing.T, gw *v1beta1.Gateway, status metav1.ConditionStatus, reason v1beta1.GatewayConditionReason) {
+	require.True(t,
+		findConditionInList(t, gw.Status.Conditions, string(v1beta1.GatewayConditionProgrammed), string(status), string(reason)),
+		"Programmed status of %s with reason %s was NOT present on Gateway %s/%s as required", status, reason, gw.Namespace, gw.Name,
+	)
+}
+
```

Comment:
Similary to what we have for [HTTPRoute](https://github.com/kubernetes-sigs/gateway-api/blob/6d8927ddeba5f08a22fd27a1fa5aa0b8b28959ee/conformance/utils/kubernetes/helpers.go#L550), can't we create a `GatewayMustHaveCondition` and pass the `Programmed` and `Accepted` conditions?

Reply:
I really liked these new ones because they had a simpler interface, however for consistency reasons I switched as requested. :+1: